rezz.lic

```diff
@@ -1,119 +1,126 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#rezz
+=end
 
 custom_require.call(%w[common common-arcana events drinfomon spellmonitor])
 
-# rezz waggle set with POM if you want
-
 class Rezz
   include DRC
   include DRCA
-
+  
   def initialize
+    DRStats.cleric? ? nil : begin echo '*** Nice try, heathen! ***'; exit end
+	
     arg_definitions = [
       [
-        { name: 'person', regex: /\w+/, description: 'Person to rezz' }
+        { name: 'player', regex: /\w+/, optional: true, description: 'Name of the player to rezz' }
       ]
     ]
-
-    @rejuv_mana = 0
+	
     args = parse_args(arg_definitions)
     @settings = get_settings
-    rezz_prep(args.person)
+	
+	args.player ? @target = DRRoom.pcs.find { |name| /^#{args.player}/i =~ name }.to_a : find_dead
+	!@target ? begin echo "*** No bodies! ***"; exit end : nil
+	
+    @rezz = if @settings.waggle_sets['rezz']['Resurrection']
+              @settings.waggle_sets['rezz']['Resurrection']
+			else
+			  echo "Using default values for REZZ, make a 'rezz'=>Resurrection waggle set to avoid this."
+			  get_data('spells').spell_data['Resurrection']
+			end
+
+    @rejuv = if @settings.waggle_sets['rezz']['Rejuvenation']
+	           @settings.waggle_sets['rezz']['Rejuvenation']
+			 else
+			   echo "Using default values for REJUV, make a 'rezz'=>Rejuvenation waggle set to avoid this."
+	           get_data('spells').spell_data['Rejuvenation']
+			 end
+
+    Flags.add('no-favor', /has no favor with h.. god!/)
+    Flags.add('rejuv-silver', 'A thin silver nimbus surrounds', 'As you concentrate, a thin silver nimbus flickers into view around')
+    Flags.add('alive', 'Nothing happens')
+    Flags.add('necro', 'has no protection from the gods. judgment')
+
+    @target.each do |name|
+      @rejuv['cast'] = "cast #{name}"
+	  rejuv_process(name)
+      Flags['alive'] ? begin echo "*** #{name} is already alive, dumbass! ***"; next end : nil
+      Flags['no-favor'] ? begin echo "*** #{name} has no favors! ***"; next end : nil
+      rezz_process(name)
+	end
   end
 
-  def rezz_prep(person)
-    # cast buffs if set
-    check_buffs
-
-    # perceive body & rejuv
-    rejuv(person)
-
-    # cast rezz
-    mana = cast_rezz
-
-    # infuse untill see spirit
-    find_soul(person, mana)
-
-    # snap cast soul bonding
-    cast_soulbonding(person)
-
-    # check for silver again. recast rejuv (or infuse again) if not
-    rejuv(person)
-
-    # gesture
-    bput("gest #{person}", 'Roundtime')
+  def find_dead
+    /You take a moment to look for everybody in the area and see (.*)/ =~ bput('look people', "You look around and notice that you're the only one in the area", /You take a moment to look for everybody in the area and see (.*)/)
+    @target = Regexp.last_match(1).sub(/ and /, ', ').scan(/the body of ([a-z[:blank:]]+) who is [a-z[:blank:]]+?,?./i).flatten.map { |obj| obj.split.last }
+	@target == [] ? @target = nil : nil
```

Comment:
So this appears that it's setting @target to nil regardless of true or false, so why the check?

Reply:
If @target is empty, it sets it to nil.  Otherwise it does nothing.

Otherwise it would send an empty array of names back and that would get past the check at line 24 or 25.  Then it would send the empty array to the @target.each do |name| section.

args.player ? @target = DRRoom.pcs.find { |name| /^#{args.player}/i =~ name }.to_a : find_dead
!@target ? begin echo "*** No bodies! ***"; exit end : nil

Reply:
Oh yes, I'm reading it. So if true do something else do nothing. This is how I would handle that, and it would follow convention with the rest of Dependency.

`@target = nil if @target.empty?`

Reply:
It sets it to nil only if true.  If false it does nothing.
What happened otherwise is that if I ran the script with no bodies in the room, it would pass an empty array to @targets and then it would get by the check on like 24 that would echo that there are no bodies and exit the script.

Reply:
Oh, that looks better.  I'll change it to that.

Reply:
Ok, I changed it to that.